testutils/infra/smtpd_mock.py

```diff
@@ -11,45 +11,48 @@
 #    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #    See the License for the specific language governing permissions and
 #    limitations under the License.
-
-import asyncore
 import base64
 import email
 import imaplib
 import os
 import pytest
 import time
-import smtpd
 import logging
 
-from threading import Thread, Condition
+from aiosmtpd.controller import Controller
+
+from threading import Condition
 
 from redo import retriable
 
 
 class Message:
-    def __init__(self, peer, mailfrom, rcpttos, subject, data):
+    def __init__(self, peer, mail_from, rcpttos, subject, data):
         self.peer = peer
-        self.mailfrom = mailfrom
+        self.mail_from = mail_from
         self.rcpttos = rcpttos
         self.subject = subject
         self.data = data
 
     def __repr__(self):
-        return f"<Message peer={self.peer} mailfrom={self.mailfrom} rcpttos={self.rcpttos} subject={self.subject}>"
+        return f"<Message peer={self.peer} mailfrom={self.mail_from} rcpttos={self.rcpttos} subject={self.subject}>"
 
 
-class SMTPServerMock(smtpd.SMTPServer):
+class SMTPServerMockHandler:
     def __init__(self, *args, **kwargs):
         self.messages = []
-        smtpd.SMTPServer.__init__(self, *args, **kwargs)
         self._msg_cond = Condition()
 
-    def process_message(self, peer, mailfrom, rcpttos, data, **kwargs):
-        self.messages.append(Message(peer, mailfrom, rcpttos, None, data))
-        logging.warning("got a message: %s" % mailfrom)
+    async def handle_DATA(self, server, session, envelope):
+        peer = session.peer
+        mail_from = envelope.mail_from
+        rcpt_tos = envelope.rcpt_tos
+        data = envelope.content
+        self.messages.append(Message(peer, mail_from, rcpt_tos, None, data))
+        logging.warning("got a message: %s" % mail_from)
```

Comment:
Should this really be a warning? I think we should downgrade to `info`

Reply:
Also, please remove the `%` construction when modernizing code. For logging it is better with:
```suggestion
        logging.warning("got a message:", mail_from)
```

Reply:
Finally, maybe this should be "got a message from:"